### ISSUE TITLE
Question of  function generate_ellipse_path_from_camera_infos()

### ISSUE THREAD

**Issue by Yuhuoo** [nerlfield/wild-gaussian-splatting#2]
Why does this function generate such high noise in the camera pose?

<img width="1314" alt="image" src="https://github.com/nerlfield/wild-gaussian-splatting/assets/91204817/3f7f813c-24a0-4372-b605-2e69c009a5d9">

The first image below is the result rendered from the training dataset, while the second one is a frame corresponding of the video rendered.

<img width="513" alt="image" src="https://github.com/nerlfield/wild-gaussian-splatting/assets/91204817/38c2a6ac-b6f6-4924-bbbb-e2fa7aa196b1">

<img width="514" alt="image" src="https://github.com/nerlfield/wild-gaussian-splatting/assets/91204817/79a0d206-2fbb-4d46-97a0-87070f4e7ec4">

**Comment by nerlfield** [comment]
Hi @Yuhuoo, thank you for your question. Could you give more details about the dataset you used? How many images are there? Are the cameras sparse?

Here is my initial guess:
I think your dataset might be small or the cameras are very sparse. This means the distance between groups of actual cameras is large, and DUST3R might not estimate camera parameters correctly. You see noisy Gaussians because this function generates new camera parameters that weren’t used in training. It seems like the Gaussians have overfitted on the gt camera parameters and haven’t generalized to new camera views, which is why you see such noisy reconstruction. In simple terms, you might need to add more intermediate images (make the dataset a little bit dense: by adding more images) for better reconstruction

**Comment by Yuhuoo** [comment]
Thank you for your reply. I wanted to ask if you have modified the code of GaussianSplatting?

It's strange that the results obtained in this project differ from the original GaussianSplatting project [https://github.com/graphdeco-inria/gaussian-splatting](url) after training, but I used the same data—point cloud and camera poses obtained from dust3r.

**Comment by Yuhuoo** [comment]
<img width="1106" alt="image" src="https://github.com/nerlfield/wild-gaussian-splatting/assets/91204817/1a4e6f37-0ffd-427d-806d-283fd8a279a2">
Why does `self.world_view_transform` need to be multiplied by 10000?

**Comment by ttsesm** [comment]
@Yuhuoo did you manage to figure out why the multiplication to `10000` is needed? We also have a similar issue with a noisy reconstruction.

**Comment by nerlfield** [comment]
I've done it to overcome the issue of numerical instability. Firstly, there's no need for concern, as the values are normalized before being used in the render function. 

The challenge arises because Dust3r outputs camera parameters in a very small range and there are values super close to 0, and these sometimes saturate to 0 after normalization before rendering. By multiplying them by 10000, the values increase and, post-normalization, they nearly reach but do not hit zero. Without this adjustment, artifacts appear in the rendered images. You may comment out this line; theoretically, it shouldn't affect the classical Gaussian splatting pipeline.

**Comment by ttsesm** [comment]
Thanks @nerlfield
